storage/client_split_test.go

```diff
@@ -412,6 +412,63 @@ func TestStoreZoneUpdateAndRangeSplit(t *testing.T) {
 	}
 }
 
+// TestMaxBytesChangeAndRangeSplit tests a scenario where a new zone config
+// changes the max bytes and triggers a range split.
+//
+// TODO(kkaneda): Tweak this test if a new range can get a new value
+// of max bytes without another zone config change.
+func TestMaxBytesChangeAndRangeSplit(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	store, stopper := createTestStore(t)
+	defer stopper.Stop()
+
+	origRng := store.LookupRange(proto.KeyMin, nil)
+
+	// Set the maxBytes and trigger a range split.
+	key := keys.MakeKey(keys.ConfigZonePrefix, proto.Key("db1"))
+	maxBytes := int64(1 << 16)
+	zoneConfig := &proto.ZoneConfig{
+		ReplicaAttrs: []proto.Attributes{
+			{},
+			{},
+			{},
+		},
+		RangeMinBytes: 1 << 8,
+		RangeMaxBytes: maxBytes,
+	}
+	if err := store.DB().Put(key, zoneConfig); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the range is split, but the max bytes has not yet been updated.
+	util.SucceedsWithin(t, time.Second, func() error {
+		newRng := store.LookupRange(proto.Key("db1"), nil)
+		if newRng.Desc().RaftID == origRng.Desc().RaftID {
+			return util.Error("expected new range created by split")
+		}
+		if newRng.GetMaxBytes() != 0 {
```

Comment:
I wasn't totally sure since the test might not be doing the right thing... @bdarnell , could you take a look? If this is a bug, will create an issue to track this.